app/services/checksum_validator.rb

```diff
@@ -2,86 +2,81 @@
-  attr_reader :bare_druid, :results, :complete_moab
+  attr_reader :druid, :results, :complete_moab
 
-  alias druid bare_druid
   delegate :moab_storage_root, to: :complete_moab
   delegate :storage_location, to: :moab_storage_root
-  delegate :mark_moab_not_found, :moab, :object_dir, :set_status_as_seen_on_disk, :update_status, to: :moab_validator
-
-  MANIFESTS = 'manifests'
-  MANIFESTS_XML = 'manifestInventory.xml'
-  MODIFIED = 'modified'
-  ADDED = 'added'
-  DELETED = 'deleted'
-  FILES = 'files'
+  delegate :moab, :object_dir, to: :moab_validator
 
   def initialize(complete_moab)
     @complete_moab = complete_moab
-    @bare_druid = complete_moab.preserved_object.druid
-    @results = AuditResults.new(bare_druid, nil, moab_storage_root, 'validate_checksums', logger: Audit::Checksum.logger)
+    @druid = complete_moab.preserved_object.druid
+    @results = AuditResults.new(druid, nil, moab_storage_root, 'validate_checksums', logger: Audit::Checksum.logger)
     # TODO: fix fragile interdependence, MoabValidator wants AuditResults instance, but we want MoabValidator#moab.current_version_id
     # in that AuditResults instance.  so set AuditResults#actual_version after both instances have been created.
     @results.actual_version = moab.current_version_id
   end
 
   def validate_checksums
     # check first thing to make sure the moab is present on disk, otherwise weird errors later
-    return results.result_array unless check_file_existence!
+    return persist! { moab_validator.mark_moab_not_found } unless moab_exists?
 
+    # These will populate the results object
     validate_manifest_inventories
     validate_signature_catalog
 
-    # This is to deal with db connection timeouts.
-    ActiveRecord::Base.clear_active_connections!
-
-    transaction_ok = ActiveRecordUtils.with_transaction_and_rescue(results) do
+    persist!(clear_connections: true) do
       complete_moab.last_checksum_validation = Time.current
       if results.result_array.empty?
         results.add_result(AuditResults::MOAB_CHECKSUM_VALID)
-        found_expected_version = moab.current_version_id == complete_moab.version
-        set_status_as_seen_on_disk(found_expected_version)
         complete_moab.update_audit_timestamps(true, true)
-        unless found_expected_version
-          results.add_result(AuditResults::UNEXPECTED_VERSION,
-                             actual_version: moab.current_version_id,
-                             db_obj_name: 'CompleteMoab',
-                             db_obj_version: complete_moab.version)
-        end
+
+        validate_versions
       else
-        update_status('invalid_checksum')
+        update_complete_moab_status('invalid_checksum')
       end
-      complete_moab.save!
     end
-    results.remove_db_updated_results unless transaction_ok
+  end
 
-    results.report_results
+  # @return [Moab::StorageObjectVersion]
+  def latest_moab_version
+    @latest_moab_version ||= moab.version_list.last
```

Comment:
presumably there is a test, or already was a test, for when moab.version_list is ... whatever it initializes to ...   Also ... is this a "complete moab"?

Or perhaps you're starting to understand why we started abbreviating ...

Reply:
This is a `Moab::StorageObjectVersion`. Renaming to make clearer.